docs/source/traitsui_user_manual/testing.rst

```diff
@@ -0,0 +1,628 @@
+
+In some situations, the GUI can be very nested and we may need to chain many
+calls to |UIWrapper.locate| before we can finally call |UIWrapper.perform| or
+|UIWrapper.inspect|::
+
+    person_pane.locate(...).locate(...).locate(...).perform(...)
+
+Examples
+========
+
+Several test examples can be found for testing
+:ref:`TraitsUI\'s own demos<traitsui-demo>`.
+
+Editors
+-------
+- :github-demo:`ButtonEditor <Standard_Editors/tests/test_ButtonEditor_demo.py>`
+- :github-demo:`ButtonEditor <Standard_Editors/tests/test_ButtonEditor_simple_demo.py>`
+- :github-demo:`CheckListEditor <Standard_Editors/tests/test_CheckListEditor_simple_demo.py>`
+- :github-demo:`EnumEditor <Standard_Editors/tests/test_EnumEditor_demo.py>`
+- :github-demo:`InstanceEditor <Standard_Editors/tests/test_InstanceEditor_demo.py>`
+- :github-demo:`ListEditor <Advanced/tests/test_List_editor_notebook_selection_demo.py>`
+- :github-demo:`TextEditor <Standard_Editors/tests/test_TextEditor_demo.py>`
+
+Applications
+------------
+- :github-demo:`Converter <Applications/tests/test_converter.py>`
+
+Debugging
+=========
+When you want to sanity check the test is doing what you intended, or when
+you want to debug an issue, the ``delay`` parameter may be useful for
+slowing down the test so that you can see the GUI being updated.
+
+Example::
+
+    from traitsui.testing.api import UITester, KeySequence
+
+    obj = App()
+    tester = UITester(delay=50)    # delay is set
+    with tester.create_ui(obj, dict(view=view)) as ui:
+        title_field = tester.find_by_name(ui, "title")
+        title_field.perform(KeySequence("New Title"))
+
+Note that there are limitations as to how truthful the GUI looks visually
+compared to manual testing.
+
+FAQ
+===
+
+.. rubric:: Is UITester GUI toolkit specific?
+
+It depends. The API of UITester is toolkit independent. As long as the behavior
+being tested is also toolkit independent, |UITester| should support test code
+with no toolkit specific logic. However if the underlying GUI behavior
+varies from toolkit to toolkit, the test code using |UITester| will need to
+adjust for that.
+
+.. rubric:: Which testing framework should I use with UITester?
+
+|UITester| does not depend on any testing framework. You should be able to use
+it with any testing framework (e.g. unittest, pytest).
+
+.. rubric:: Is UITester compatible with PyFace ModalDialogTester?
+
+Yes. For example, you can use |UITester| to launch a modal dialog, and use
+ModelDialogTester to close it.
+
+.. rubric:: Is UITester compatible with PyFace GuiTestAssistant?
+
+Yes.
+
+|UIWrapper.perform| and |UIWrapper.inspect| automatically request GUI
+events to be processed. Where they are used entirely for modifying GUI states,
+you may find uses of some of GuiTestAssistant features no longer necessary.
+
+.. rubric:: I use the delay parameter to play back my test but it does not look identical to when I test the GUI manually, why?
```

Comment:
is it not possible to wrap `.. rubric::` lines?

Reply:
It is possible actually. Fixing...